File: frontend/src/types/idea.ts
export interface VideoIdea {
  title: string;
  segments_est: number;
  description: string;
  keywords: string[];
  cold_open_text?: string;
  format_id?: string;
  closing_image?: string;
}

export interface GenerateIdeasResponse {
  ideas: VideoIdea[];
}

export type IdeaStatus = "idea" | "in_progress" | "scripted" | "published";
export type IdeaSource = "manual" | "for_you" | "trending";
export type ColdOpenStatus = "pending" | "generating" | "ready" | "refining" | "scored" | "failed";

export interface Idea {
  id: string;
  text: string;
  description: string;
  category: string;
  rank: number;
  status: IdeaStatus;
  source: IdeaSource;
  cold_open_status: ColdOpenStatus;
  cold_open_variants_json: string;
  selected_hook_json: string;
  hook_score: number | null;
  hook_score_json: string;
  created_at: string;
}

File: backend/models/idea.py
"""Idea model for persistent video idea notes with priority ranking and hook scoring."""

import uuid
from datetime import datetime, timezone

from sqlmodel import Field, SQLModel


VALID_STATUSES = {"idea", "in_progress", "scripted", "published"}
VALID_SOURCES = {"manual", "for_you", "trending"}
VALID_COLD_OPEN_STATUSES = {"pending", "generating", "ready", "refining", "scored", "failed"}


class Idea(SQLModel, table=True):
    __tablename__ = "ideas"

    id: str = Field(default_factory=lambda: uuid.uuid4().hex, primary_key=True)
    text: str
    description: str = Field(default="")
    category: str = Field(default="")
    rank: int = Field(default=50)
    status: str = Field(default="idea")
    source: str = Field(default="manual")
    cold_open_status: str = Field(default="pending")
    cold_open_job_id: str = Field(default="")
    cold_open_variants_json: str = Field(default="")
    selected_hook_json: str = Field(default="")
    hook_score: int | None = Field(default=None)
    hook_score_json: str = Field(default="")
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
